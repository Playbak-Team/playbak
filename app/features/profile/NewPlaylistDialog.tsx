import { Dialog } from '@material-ui/core';
import Button from '@material-ui/core/Button/Button';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText/DialogContentText';
import TextField from '@material-ui/core/TextField/TextField';
import React, { useState } from 'react';
import { NewPlaylistDialogProps } from '../../types';

export default function NewPlayListDialog(props: NewPlaylistDialogProps) {
  const { playlistOpen, setPlaylistOpen } = props;
  const [enteredValue, setEnteredValue] = useState(() => {
    const savedValue = window.localStorage.getItem('playlist');

    return savedValue !== null ? savedValue : '';
  });

  function handleClose(value: string) {
    if (value !== '') window.localStorage.setItem('playlist', value);
    setPlaylistOpen(false);
  }

  return (
    <Dialog open={playlistOpen} aria-labelledby="form-dialog-title">
      <DialogContent>
        <DialogContentText>Enter playlist ID</DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          id="name"
          label="Playlist"
          fullWidth
          value={enteredValue}
          onChange={(e) => setEnteredValue(e.target.value)}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={() => handleClose(enteredValue)} color="primary">
          Create
        </Button>
        <Button onClick={() => handleClose('')} color="primary">
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
}
