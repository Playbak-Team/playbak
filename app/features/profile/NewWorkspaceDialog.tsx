import { Dialog } from '@material-ui/core';
import Button from '@material-ui/core/Button/Button';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText/DialogContentText';
import TextField from '@material-ui/core/TextField/TextField';
import React, { useState } from 'react';
import { NewWorkspaceDialogProps } from '../../types';

export default function NewWorkspaceDialog(props: NewWorkspaceDialogProps) {
  const { isWkOpen, handleWkClose } = props;
  const [enteredName, setEnteredName] = useState('');

  function handleClose(value: string) {
    handleWkClose(value);
    setEnteredName('');
  }

  return (
    <Dialog open={isWkOpen} aria-labelledby="form-dialog-title">
      <DialogContent>
        <DialogContentText>
          Enter a name for this new workspace
        </DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          id="name"
          label="Workspace"
          fullWidth
          value={enteredName}
          onChange={(e) => setEnteredName(e.target.value)}
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={() => handleClose(enteredName)} color="primary">
          Create
        </Button>
        <Button onClick={() => handleClose('')} color="primary">
          Cancel
        </Button>
      </DialogActions>
    </Dialog>
  );
}
