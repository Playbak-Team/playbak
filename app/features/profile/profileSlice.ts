import { createSlice, PayloadAction } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
import { AppThunk, RootState } from '../../store';
import { VideoStateInterface } from '../../interfaces';

const initialState: VideoStateInterface = {
  videoURLS: [
    'C:\\Users\\kevin\\Desktop\\cs341lectures',
    'C:\\Users\\kevin\\Desktop\\cs349lectures',
  ],
  currentVideo: '',
  snackbarActive: false,
  snackbarMessage: '',
  snackbarSeverity: undefined,
};

// 1. Create a workspace (ie. F20, W21)
// 2. Create a course
// - Name
// - Create a directory with
// - Videos, Assignments, Notes, Others folders
// - maybe can add cache of some sort here as well
// 3. Select a workspace

const videoSlice = createSlice({
  name: 'video',
  initialState,
  reducers: {
    disableSnackbar: (state) => {
      state.snackbarActive = false;
    },
    addToURLS: (state, action: PayloadAction<string>) => {
      if (!state.videoURLS.includes(action.payload) && action.payload !== '') {
        state.videoURLS.push(action.payload);
        state.snackbarMessage = `Added ${action.payload} to paths`;
        state.snackbarSeverity = 'success';
      } else if (state.videoURLS.includes(action.payload)) {
        state.snackbarMessage = 'Path already exists';
        state.snackbarSeverity = 'error';
      } else if (action.payload === '') {
        state.snackbarMessage = 'Cannot add an empty path';
        state.snackbarSeverity = 'error';
      }
      state.snackbarActive = true;
    },
    setVideo: (state, action: PayloadAction<string>) => {
      state.currentVideo = action.payload;
      state.snackbarActive = true;
      state.snackbarMessage = `Video set to ${action.payload}`;
      state.snackbarSeverity = 'success';
    },
  },
});

export const { addToURLS, setVideo, disableSnackbar } = videoSlice.actions;

export default videoSlice.reducer;

export const getPathURLS = (state: RootState) => state.video.videoURLS;

export const getCurrentVideo = (state: RootState) => state.video.currentVideo;

export const isSnackBarActive = (state: RootState) =>
  state.video.snackbarActive;

export const getSnackBarMessage = (state: RootState) =>
  state.video.snackbarMessage;

export const getSnackBarSeverity = (state: RootState) =>
  state.video.snackbarSeverity;
