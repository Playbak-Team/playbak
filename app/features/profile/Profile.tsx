import React, { useState, useEffect, useCallback } from 'react';
import {
  makeStyles,
  createStyles,
  Theme,
  withStyles,
} from '@material-ui/core/styles';
import { useSelector, useDispatch } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import AccountBoxIcon from '@material-ui/icons/AccountBox';
import IconButton from '@material-ui/core/IconButton';
import CheckIcon from '@material-ui/icons/Check';
import EditIcon from '@material-ui/icons/Edit';
import TextField from '@material-ui/core/TextField';
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import { ipcRenderer } from 'electron';
import NewWorkspaceDialog from './NewWorkspaceDialog';
import WorkspaceInterface from './WorkspaceInterface';
import {
  getCurrentTerm,
  setName,
  getName,
  getAllWorkspaces,
  addWorkspace,
  setWorkspace,
  setCourses,
} from './profileSlice';
import {
  showSuccess,
  showError,
} from '../../components/Snackbar/snackBarSlice';
import { WorkspaceEntryProps } from '../../types';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      overflowX: 'hidden',
      overflowY: 'hidden',
      backgroundColor: '#0B1E38',
      height: '100vh',
    },
    paper: {
      padding: theme.spacing(2),
      textAlign: 'center',
      color: 'white',
      minHeight: '100vh',
      display: 'flex',
      flexDirection: 'column',
      background: '#00011a',
      boxShadow:
        '8px 0px 0px 0px #DCD0C0, 0px 8px 0px 0px #B1938B, -8px 0px 0px 0px #4E4E56, 0px 0px 0px 8px #DA635D, 5px 5px 15px 5px rgba(0,0,0,0)',
    },
    row: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
    },
    namerow: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'space-between',
      alignItems: 'center',
      borderBottom: '3px solid #7FC5DC',
      paddingBottom: '10px',
    },
    termrow: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      marginTop: '3vh',
      borderBottom: '3px solid #7FC5DC',
      paddingBottom: '3vh',
    },
    workspacesdivparent: {
      minHeight: '50vh',
      maxHeight: '50vh',
      width: '100%',
      overflow: 'hidden',
      height: '100%',
      position: 'relative',
    },
    workspacesdiv: {
      position: 'absolute',
      top: '0px',
      bottom: '0px',
      left: '0px',
      right: '-17px',
      overflowY: 'scroll',
    },
    workspacestitle: {
      display: 'flex',
      flexDirection: 'row',
      marginTop: '2vh',
      marginBottom: '2vh',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
    workspaceentry: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alginItems: 'center',
      margin: '15px',
      padding: '10px',
      backgroundColor: '#3f2f4f',
      boxShadow: '0px 10px 13px -7px #000000, 5px 5px 15px 5px rgba(0,0,0,0)',
    },
    whitetext: {
      color: 'white',
    },
  })
);

const CssTextField = withStyles({
  root: {
    '& label.Mui-focused': {
      color: 'white',
    },
    '& .MuiInput-underline:after': {
      borderBottomColor: 'white',
    },
    '& .MuiInputBase-input': {
      color: 'white',
    },
    '& .MuiFormLabel-root': {
      color: 'white',
    },
    '& .MuiOutlinedInput-root': {
      '& fieldset': {
        borderColor: 'white',
      },
      '&:hover fieldset': {
        borderColor: 'white',
      },
      '&.Mui-focused fieldset': {
        borderColor: 'white',
      },
    },
    color: 'white',
  },
})(TextField);

const WorkspaceEntry = React.memo(function WorkspaceE(
  props: WorkspaceEntryProps
) {
  const classes = useStyles();
  const dispatch = useDispatch();
  const { name } = props;
  return (
    <div className={classes.workspaceentry}>
      <p>{name}</p>
      <IconButton
        style={{ color: 'yellow' }}
        onClick={() => {
          dispatch(setWorkspace(name));
        }}
      >
        <ArrowForwardIosIcon />
      </IconButton>
    </div>
  );
});

export default function Profile() {
  const [editingName, setEditingName] = useState(false);
  const [isWkOpen, setIsWkOpen] = useState(false);
  const classes = useStyles();
  const name = useSelector(getName);
  const allWorkspaces = useSelector(getAllWorkspaces);
  const currentTerm = useSelector(getCurrentTerm);
  const [nameChange, setNameChange] = useState('');
  const dispatch = useDispatch();
  const setEditingNameTrue = useCallback(() => setEditingName(true), []);

  function handleWkClose(value: string) {
    if (value !== '' && !allWorkspaces.includes(value)) {
      dispatch(addWorkspace(value));
      dispatch(
        showSuccess(`The workspace ${value} has been successfully added`)
      );
    } else {
      dispatch(showError(`The workspace ${value} already exists`));
    }
    setIsWkOpen(false);
  }

  useEffect(() => {
    ipcRenderer.on('return-courses', (_, courses: string[]) => {
      dispatch(setCourses(Object.values(courses)));
    });
    return () => {
      ipcRenderer.removeAllListeners('return-courses');
    };
  }, [dispatch]);

  return (
    <div className={classes.root}>
      <Grid container spacing={3}>
        <Grid item xs={3}>
          <div className={classes.paper}>
            <div className={classes.namerow}>
              <AccountBoxIcon style={{ fontSize: 150 }} />
              {editingName ? (
                <div>
                  <CssTextField
                    id="new-name-field"
                    label="Enter new name"
                    defaultValue={name}
                    variant="outlined"
                    InputProps={{
                      className: classes.whitetext,
                    }}
                    onChange={(e) => setNameChange(e.target.value)}
                    value={nameChange}
                  />
                  <IconButton
                    style={{ fontSize: 60, color: 'green' }}
                    aria-label="edit name confirm"
                    id="edit-name-confirm"
                    onClick={() => {
                      dispatch(setName(nameChange));
                      setEditingName(false);
                    }}
                  >
                    <CheckIcon />
                  </IconButton>
                </div>
              ) : (
                <div style={{ display: 'flex', flexDirection: 'row' }}>
                  <h2 id="user-name">{name}</h2>
                  <IconButton
                    color="secondary"
                    style={{ fontSize: 60 }}
                    aria-label="edit name"
                    id="edit-name"
                    onClick={setEditingNameTrue}
                  >
                    <EditIcon />
                  </IconButton>
                </div>
              )}
            </div>
            <div className={classes.termrow}>
              <div style={{ display: 'flex', flexDirection: 'row' }}>
                <div style={{ marginRight: '1vw' }}>Selected Term:</div>
                {currentTerm === '' ? (
                  <div>&nbsp;NONE</div>
                ) : (
                  <div id="current-term">{currentTerm}</div>
                )}
              </div>
              <Button
                variant="contained"
                size="small"
                color="secondary"
                onClick={() => setIsWkOpen(true)}
                style={{ maxWidth: 'max-content' }}
              >
                Add a new workspace
              </Button>
            </div>
            <NewWorkspaceDialog
              isWkOpen={isWkOpen}
              handleWkClose={handleWkClose}
            />
            <div className={classes.workspacestitle}>Available Workspaces</div>
            <div className={classes.workspacesdivparent}>
              <div className={classes.workspacesdiv}>
                {allWorkspaces.map((n) => (
                  <div key={n}>
                    <WorkspaceEntry key={n} name={n} />
                  </div>
                ))}
              </div>
            </div>
          </div>
        </Grid>
        <Grid item xs={9}>
          <WorkspaceInterface workspace={currentTerm} />
        </Grid>
      </Grid>
    </div>
  );
}
