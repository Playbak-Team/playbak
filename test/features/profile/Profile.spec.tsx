/* eslint react/jsx-props-no-spreading: off, @typescript-eslint/ban-ts-comment: off */
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import Profile from '../../../app/features/profile/Profile';
import * as profileSlice from '../../../app/features/profile/profileSlice';

Enzyme.configure({ adapter: new Adapter() });
jest.useFakeTimers();

function setup(
  preloadedState: {
    profile: {
      name: string;
      selectedWorkspace: string;
      availableWorkspaces: string[];
      courses: string[];
      links: string[];
    };
  } = {
    profile: {
      name: 'bob',
      selectedWorkspace: 'F20',
      availableWorkspaces: ['F20', 'W21', 'S21', 'F21'],
      courses: ['CS341', 'CS350'],
      links: [],
    },
  }
) {
  const store = configureStore({
    reducer: { profile: profileSlice.default },
    preloadedState,
  });

  const getWrapper = () =>
    mount(
      <Provider store={store}>
        <Router>
          <Profile />
        </Router>
      </Provider>
    );
  const component = getWrapper();
  return {
    store,
    component,
  };
}

describe('Counter component', () => {
  it('should match exact snapshot', () => {
    const { store } = setup();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Profile />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
