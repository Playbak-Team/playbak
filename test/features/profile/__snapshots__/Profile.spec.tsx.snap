// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should match exact snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      <div
        className="makeStyles-paper-2"
      >
        <div
          className="makeStyles-namerow-4"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            style={
              Object {
                "fontSize": 100,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
            />
          </svg>
          <div>
            <h2>
              bob
              <button
                aria-label="edit name"
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={
                  Object {
                    "fontSize": 60,
                  }
                }
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
              </button>
            </h2>
          </div>
        </div>
        <div
          className="makeStyles-termrow-5"
        >
          Current Selected Term:
          <div>
            Â 
            F20
          </div>
        </div>
        <div
          className="makeStyles-workspacestitle-8"
        >
          Available Workspaces
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root makeStyles-buttonstyle-13"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="MuiCollapse-container MuiCollapse-hidden"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginBottom": "10px",
                  }
                }
              >
                <div
                  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-14"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink={false}
                    htmlFor="outlined-basic"
                    id="outlined-basic-label"
                  >
                    Enter workspace name
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root makeStyles-whitetext-12 MuiInputBase-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      id="outlined-basic"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-17"
                      >
                        <span>
                          Enter workspace name
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  aria-label="edit"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "green",
                      "fontSize": 100,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-workspacesdivparent-6"
        >
          <div
            className="makeStyles-workspacesdiv-7"
          >
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                F20
                <button
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                W21
                <button
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                S21
                <button
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                F21
                <button
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
    >
      <div
        className="makeStyles-workspacearea-10"
      >
        <div
          className="makeStyles-workspaceareatitle-11"
        >
          <div>
            F20
          </div>
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root makeStyles-buttonstyle-13"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          className="MuiCollapse-container MuiCollapse-hidden"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "marginBottom": "10px",
                  }
                }
              >
                <div
                  className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-14"
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                >
                  <label
                    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                    data-shrink={false}
                    htmlFor="outlined-basic"
                    id="outlined-basic-label"
                  >
                    Enter path
                  </label>
                  <div
                    className="MuiInputBase-root MuiOutlinedInput-root makeStyles-whitetext-12 MuiInputBase-formControl"
                    onClick={[Function]}
                  >
                    <input
                      aria-invalid={false}
                      autoFocus={false}
                      className="MuiInputBase-input MuiOutlinedInput-input"
                      disabled={false}
                      id="outlined-basic"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden={true}
                      className="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        className="PrivateNotchedOutline-legendLabelled-17"
                      >
                        <span>
                          Enter path
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  aria-label="edit"
                  className="MuiButtonBase-root MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "color": "green",
                      "fontSize": 100,
                    }
                  }
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="makeStyles-workspacesdivparent-6"
        >
          <div
            className="makeStyles-workspacesdiv-7"
          >
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                CS341
              </div>
            </div>
            <div>
              <div
                className="makeStyles-workspaceentry-9"
              >
                CS350
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
